Debian 13

evgeni opened the pull request.

evgeni commented on development/playbooks/setup-repositories/setup-repositories.yaml:
```diff
@@ -9,10 +9,30 @@
+      community.general.copr:
+        host: copr.fedorainfracloud.org
+        state: enabled
+        name: evgeni/hammer-el10
```
you may know this copr from fixes like https://github.com/theforeman/foreman-packaging/pull/12499 and https://github.com/theforeman/foreman-packaging/pull/12500

evgeni replied:
(this needs to be read in the voice of Troy McClure)

ekohl commented on development/playbooks/setup-repositories/setup-repositories.yaml:
```diff
@@ -9,10 +9,23 @@
       vars:
         foreman_puppet_repositories_version: "8"
+      when:
+        - ansible_distribution_major_version == '9'
```
Should this be when the certificate source is the installer instead?

evgeni replied:
Yepp, but this detail is not passed to the "setup repos" playbook, and I was lazy

evgeni committed:
add Debian 13 to the mix

evgeni committed:
Debian needs to install netavark explicitly

evgeni committed:
python3-libsemanage is called python3-semanage on Debian

evgeni committed:
make httpd role work on Debian

evgeni committed:
skip hammer on Debian

ekohl commented on src/roles/httpd/tasks/main.yml:
```diff
@@ -1,20 +1,28 @@
+    - ansible_selinux.status == 'enabled'
+
+- name: Enable required modules
+  community.general.apache2_module:
```
We may also want to enable the event MPM in a similar way.

evgeni replied:
Mostly because it's a dependency we don't yet have

ekohl replied:
It also mentions compatibility:
> This does not work on RedHat-based distributions. It does work on Debian- and SuSE-based distributions. Whether it works on others depend on whether the C(a2enmod) and C(a2dismod) tools are available or not.

So configuring Apache with Ansible is certainly worse than what we had with our Puppet module.

evgeni commented on src/roles/httpd/tasks/main.yml:
```diff
@@ -45,11 +53,11 @@
-    name: httpd
-    state: started
+    name: "{{ httpd_service }}"
+    state: restarted
```
There is https://github.com/theforeman/foremanctl/issues/92

But we could make it broader

evgeni committed:
Refresh package cache on Debian during setup-repositories

evgeni committed:
disable synced folder, takes time, we don't use it

evgeni committed:
Debian doesn't ship curl in their box

jakduch commented:
Hi @evgeni, great work on the Debian 13 support! I built on top of your branch and found a few things that could be improved. I created PR #427 targeting your `debian` branch with the following commits:

**Bug fixes:**
- Fix duplicate `conf.d/` in `foreman-vhost.conf.j2` — `httpd_conf_path` already contains `conf.d` (RedHat) / `conf-enabled` (Debian), so the extra `conf.d/` segment in the `IncludeOptional` path on line 5 results in an invalid path. The SSL vhost was already correct.
- Fix spurious space in `cleanup.yml` line 4: `{{ httpd_conf_path}} /05-` should be `{{ httpd_conf_path }}/05-`

**Debian adaptations:**
- Make httpd IPA external auth OS-aware — use `libapache2-mod-*` packages and `community.general.apache2_module` (a2enmod/a2dismod) on Debian instead of manual LoadModule conf files
- Add OS-specific vars for hammer role (`ruby-hammer-cli-*` on Debian vs `rubygem-hammer_cli_*` on RedHat) and load them via `include_vars`
- Add Foreman repository setup for Debian in the development playbook (using `theforeman.operations.foreman_repositories`)
- Enable hammer deployment on Debian in CI (remove the centos-only condition)

Feel free to cherry-pick any of these commits directly into your branch, or merge #427 if you prefer. Happy to adjust anything!

evgeni committed:
Only set Policy on non-Debian

This needs Podman 5.6, while Debian ships 5.4.

https://github.com/podman-container-tools/podman/commit/439448c2873246e71591df4daddf4dfe2bd04f2e

evgeni commented on src/roles/httpd/tasks/main.yml:
```diff
@@ -117,6 +123,6 @@
-    name: httpd
+    name: "{{ httpd_service }}"
     state: started
     enabled: true
```
Services are enabled by default on Debian, so this line doesn't trigger an "enable" action, and thus the "Add httpd to foreman.target" is never applied (as `systemctl enable` is where the symlinks are created for the targets). 😿
